.github/workflows/python.yml

```diff
@@ -8,6 +8,8 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+     - name: Setup Python
+       uses: actions/setup-python@v5
```

Comment:
```suggestion
       - name: Setup Python
         uses: actions/setup-python@v5
         with:
           cache: 'pip'
```

Reply:
For chacing pip, it would be better to use requirements.txt instead of direct `pip install`.

Reply:
Is that stored in the repo's root or under `.github`? (sorry I tried to google this but did not find anything)

Reply:
It could be anywhere but the address is based on the repo's root. For example, it may be saved in the .github folder, and used like this:
`pip install -r .github/requirements.txt`

Reply:
In general, we could also use the [super-linter](https://github.com/marketplace/actions/super-linter) which I did for WALLE, but I did not want to change too much in the beginning and just fix the action.